src/internal/kopia/upload.go

```diff
@@ -347,13 +410,20 @@ func buildKopiaDirs(dirName string, dir *treeMap, progress *corsoProgress) (fs.D
 
 	return virtualfs.NewStreamingDirectory(
 		encodeAsPath(dirName),
-		getStreamItemFunc(childDirs, dir.collection, progress),
+		getStreamItemFunc(childDirs, dir.collection, dir.baseDir, progress),
 	), nil
 }
 
 type treeMap struct {
-	childDirs  map[string]*treeMap
+	// Child directories of this directory.
+	childDirs map[string]*treeMap
+	// Reference to data pulled from the external service. Contains only items in
+	// this directory. Does not contain references to subdirectories.
 	collection data.Collection
+	// Reference to directory in base snapshot. Can contain items and
+	// subdirectories but the subdirectories should be added to childDirs and
+	// ignored when handing items to kopia during the upload process.
```

Comment:
This is a pretty dense explanation, and somewhat confusing.  There's space to stretch it out another sentence or two and provide specific details about there wheres and whys it's describing.